cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/data/CassandraDataLayer.java

```diff
@@ -343,7 +346,7 @@ private CompletionStage<Map<String, AvailabilityHint>> createSnapshot(RingRespon
                                 snapshotName, keyspace, table, datacenter, ringEntry.fqdn());
                     SidecarInstance sidecarInstance = new SidecarInstanceImpl(ringEntry.fqdn(), sidecarClientConfig.effectivePort());
                     createSnapshotFuture = sidecar
-                                           .createSnapshot(sidecarInstance, keyspace, table, snapshotName)
+                                           .createSnapshot(sidecarInstance, maybeQuotedKeyspace, maybeQuotedTable, snapshotName)
```

Comment:
if the bridge is initialized correctly, the fields will never be null when we reach this point.

Reply:
however, the cleanup of snapshots happens on a different thread, if an exception occurs before the bridge has been initialized (i.e an error occurs when getting the cassandra version `String cassandraVersion = getEffectiveCassandraVersionForRead(clusterConfig, nodeSettings);` ) then the shutdown hook will be triggered and it will try to cleanup snapshots, even though the snapshot was never created in the first place